modules/core/src/test/java/org/apache/ignite/internal/client/thin/ThinClientPartitionAwarenessStableTopologyTest.java

```diff
@@ -409,14 +600,29 @@ private void testNotApplicableCache(String cacheName) {
      * @param keyFactory Key factory function.
      */
     private void testApplicableCache(String cacheName, Function<Integer, Object> keyFactory) throws Exception {
+        testApplicableCache(cacheName, keyFactory, true);
+    }
+
+    /**
+     * @param cacheName Cache name.
+     * @param keyFactory Key factory function.
+     * @param partitionsRequestExpected If {@code true}, awaits {@link ClientOperation#CACHE_PARTITIONS} first.
+     */
+    private void testApplicableCache(
+        String cacheName,
+        Function<Integer, Object> keyFactory,
+        boolean partitionsRequestExpected
+    ) throws Exception {
         ClientCache<Object, Object> clientCache = client.cache(cacheName);
         IgniteInternalCache<Object, Object> igniteCache = grid(0).context().cache().cache(cacheName);
 
         clientCache.put(keyFactory.apply(0), 0);
 
-        TestTcpClientChannel opCh = affinityChannel(keyFactory.apply(0), igniteCache);
+        TestTcpClientChannel opCh = affinityChannel(0, igniteCache);
```

Comment:
`keyFactory.apply(0)` - correct implementation, why it was changed?

Reply:
Hm. Yes. Confused with a key->value factory